[TensorExpr] Minor cleanup in TensorExprKernel::computeValue

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60041 [TensorExpr] Minor cleanup in TensorExprKernel::computeValue**



Differential Revision: [D29146709](https://our.internmc.facebook.com/intern/diff/D29146709)

Commit:
[TensorExpr] Minor cleanup in TensorExprKernel::computeValue

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

navahgar commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -2778,30 +2778,18 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
-      }
-      auto outputType = findDtypeForValue(v->node()->output());
-      std::vector<ExprHandle> outputShape = {};
-      // shape inference not implemented for sum
```
> // shape inference not implemented for sum

Do you know why this was done before? Is it okay to remove this check now, as your change is doing?

ZolotukhinM replied:
We're not using shape inference anyway (ie. `aten::sum` is not any special here), the test was just missing `requires_grad=0` in its listing. Previously we did use TE shape inference and that's probably this check was added.

navahgar reviewed (approved):
LGTM

Commit:
rebase on "[TensorExpr] Minor cleanup in TensorExprKernel::computeValue"


Differential Revision: [D29146709](https://our.internmc.facebook.com/intern/diff/D29146709)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

Commit:
Update on "[TensorExpr] Minor cleanup in TensorExprKernel::computeValue"


Differential Revision: [D29146709](https://our.internmc.facebook.com/intern/diff/D29146709)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

Commit:
rebase on "[TensorExpr] Minor cleanup in TensorExprKernel::computeValue"




Differential Revision: [D29146709](https://our.internmc.facebook.com/intern/diff/D29146709)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

Commit:
rebase on "[TensorExpr] Minor cleanup in TensorExprKernel::computeValue"




Differential Revision: [D29146709](https://our.internmc.facebook.com/intern/diff/D29146709)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

Commit:
rebase on "[TensorExpr] Minor cleanup in TensorExprKernel::computeValue"




Differential Revision: [D29146709](https://our.internmc.facebook.com/intern/diff/D29146709)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).

Commit:
rebase on "[TensorExpr] Minor cleanup in TensorExprKernel::computeValue"




Differential Revision: [D29146709](https://our.internmc.facebook.com/intern/diff/D29146709)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29146709).
